src/firefox/lib/glue.js

```diff
@@ -170,6 +180,58 @@ function setUpConnection(freedom, panel, button) {
       });
     }
   });
+
+
+  // Check if user already has a tab open to the uProxy install page.
+  for (var tab of tabs) {
+    if (isInstallPage(tab.url)) {
+      // Attach our content script to the existing tab.
+      tab.attach({contentScriptFile: contentScriptFile});
+
+      emitPromoIfFound(tab.url);
+    }
+  }
+
+  // Attach a handler to check if a tab is opened in the future with a promo.
+  tabs.on('pageshow', function (tab) {
+    emitPromoIfFound(tab.url);
```

Comment:
@jpevarnek When you're ready to end-to-end test the promo changes, maybe we can look at this together.